File: Ejercicio.py
Enteros = [2,4,6,8,10,12,4,9,5,3,7,1]
print(Enteros)


def detecrar_orden(Orden):
    asc = all(Orden[i] <= Orden[i + 1] for i in range(len(Orden) - 1))
    desc = all(Orden[i] >= Orden[i + 1] for i in range(len(Orden) - 1))
    if asc:
     print("El arreglo esta en orden ascendente")
    elif desc:
     print("El arreglo esta en orden descendente")      
    else:
     print("El arreglo no esta ordenado")   
   
detecrar_orden(Enteros)


def detectar_progresion(arr):
    if len(arr) < 2:
        print("No se puede determinar con menos de 2 elementos")
        return
    
    # Verificar progresión aritmética
    d = arr[1] - arr[0]
    es_aritmetica = all(arr[i+1] - arr[i] == d for i in range(len(arr)-1))
    
    # Verificar progresión geométrica (evitar división por cero)
    if arr[0] != 0:
        r = arr[1] / arr[0]
        es_geometrica = all(arr[i+1] / arr[i] == r for i in range(len(arr)-1) if arr[i] != 0)
    else:
        es_geometrica = False
    
    if es_aritmetica:
        print("El arreglo forma una progresión aritmética")
    elif es_geometrica:
        print("El arreglo forma una progresión geométrica")
    else:
        print("El arreglo no forma una progresión")

detectar_progresion(Enteros)

File: TypeScript/Ejercicio.ts
let arreglos: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
console.log(arreglos);

function detectar_orden(arr: number[]): string {
    let acd = true;
    let dec = true;

    for (let i = 0; i < arr.length - 1; i++) {
        if (arr[i] > arr[i + 1]) {
            acd = false;
        }
        if (arr[i] < arr[i + 1]) {
            dec = false;
        }
    }

    if (acd) {
        return "El arreglo está en orden ascendente";
    } else if (dec) {
        return "El arreglo está en orden descendente";
    } else {
        return "El arreglo no está en orden";
    }
}

console.log(detectar_orden(arreglos));

function detectar_progresion(arr: number[]): string {
    if (arr.length < 2) {
        return "El arreglo es demasiado corto para determinar la progresión";
    }

    
    const diferencia = arr[1] - arr[0];
    let esAritmetica = true;
    for (let i = 1; i < arr.length - 1; i++) {
        if (arr[i + 1] - arr[i] !== diferencia) {
            esAritmetica = false;
            break;
        }
    }

   
    let esGeometrica = true;
    if (arr[0] === 0) {
        esGeometrica = false;
    } else {
        const razon = arr[1] / arr[0];
        for (let i = 1; i < arr.length - 1; i++) {
            if (arr[i] === 0 || arr[i + 1] / arr[i] !== razon) {
                esGeometrica = false;
                break;
            }
        }
    }

    if (esAritmetica) {
        return "El arreglo es una progresión aritmética";
    } else if (esGeometrica) {
        return "El arreglo es una progresión geométrica";
    } else {
        return "El arreglo no es una progresión aritmética ni geométrica";
    }
}

console.log(detectar_progresion(arreglos));